src/main/java/com/rapid7/client/dcerpc/objects/RPCUnicodeString.java

```diff
@@ -200,39 +196,33 @@ public void unmarshalDeferrals(PacketInput in) throws IOException {
             //Preamble
             // <NDR: unsigned long> MaximumCount for conformant array - This is *not* the size of the array, so is not useful to us
             in.align(Alignment.FOUR);
-            in.readInt();
+            in.fullySkipBytes(4);
 
             //Entity
             // <NDR: unsigned long> Offset for varying array
             // Alignment: 4 - Already aligned
-            final int offset = in.readInt();
+            final int offset = readIndex(in);
             // <NDR: unsigned long> ActualCount for varying array
             // Alignment: 4 - Already aligned
-            final int actualCount = in.readInt();
+            final int actualCount = readIndex(in);
```

Comment:
Q. Did you find any case passing argument with singed int (i.e.negative value)?
Anyway, to remove signed int you might add the bit & operation(FFFF FFFF) for safety?

Reply:
`readIndex` was implemented here: https://github.com/rapid7/smbj-rpc/pull/57/files#diff-8390af140f5b6f61873bcbec7f8cd8d2R255

It uses `PacketInput.readUnsignedInt()`, which will use `& 0xFFFFFFFF` as you are suggesting: https://github.com/rapid7/smbj-rpc/pull/57/files#diff-bbe35e255b3d87886cd9524f169d0291R99

We should never get a negative number in the response this way, as we never interpret it that way. :+1:

Reply:
cool !